wallet/src/main/scala/org/bitcoins/wallet/internal/TransactionProcessing.scala

```diff
@@ -461,6 +461,8 @@ private[bitcoins] trait TransactionProcessing extends WalletLogger {
       transaction: Transaction,
       blockHashOpt: Option[DoubleSha256DigestBE],
       foundTxo: SpendingInfoDb): Future[SpendingInfoDb] = {
+    logger.info(
+      s"processExistingReceivedTxo txId=${transaction.txIdBE.hex} foundingTxo.outPoint=${foundTxo.outPoint}")
```

Comment:
These txids end up being the same, this seems wrong to me?

```
10:53:25.467 INFO Wallet$WalletImpl - processExistingReceivedTxo txId=6d6e8213a449d5797cc3b629f41ce075b30f626a44478e1f071f10a7c9f3199e foundingTxo.outPoint=TransactionOutPoint(6d6e8213a449d5797cc3b629f41ce075b30f626a44478e1f071f10a7c9f3199e:0)
10:53:25.468 INFO Wallet$WalletImpl - processExistingReceivedTxo txId=6d6e8213a449d5797cc3b629f41ce075b30f626a44478e1f071f10a7c9f3199e foundingTxo.outPoint=TransactionOutPoint(6d6e8213a449d5797cc3b629f41ce075b30f626a44478e1f071f10a7c9f3199e:2)
10:53:25.470 INFO Wallet$WalletImpl - processExistingReceivedTxo txId=6d6e8213a449d5797cc3b629f41ce075b30f626a44478e1f071f10a7c9f3199e foundingTxo.outPoint=TransactionOutPoint(6d6e8213a449d5797cc3b629f41ce075b30f626a44478e1f071f10a7c9f3199e:3)
```

Reply:
That's right the tx is the transaction we received in. The outpoint pointing to which output we received with in the transaction. It doesn't have anything to do with spending here

Reply:
This isn't right, how can the `spendingTxId` be the same as the `outpoint` txid? You can't have a self referential transaction like that. If we are _receiving_ funds, how can we have a _spending_ txid?